profiler/src/ProfilerEngine/Datadog.Profiler.Native/ClrEventsParser.cpp

```diff
@@ -0,0 +1,347 @@
+
+        // Contention events
+        case  81: return "ContentionStart";
+        case  91: return "ContentionStop";
+
+        // other verbose
+        //
+        default:
+        {
+            return nullptr;
+        }
+    }
+}
+
+void DumpEvent(
+    std::string& providerName,
+    const WCHAR* name,
+    DWORD id,
+    DWORD version,
+    INT64 keywords,
+    DWORD eventId,
+    DWORD eventVersion,
+    ULONG numStackFrames
+    )
+{
+    std::stringstream buffer;
+    buffer << providerName << " | "
+           << "k=0x" << std::hex << keywords << std::dec << " = ";
+
+    if ((name != nullptr) && WStrLen(name) > 0)
+    {
+        buffer << "'" << shared::ToString(shared::WSTRING(name)) << "' : ";
+    }
+    else
+    {
+        // most event name are empty strings
+        const char* wellKnownEventName = GetEventName(id);
+        if (wellKnownEventName != nullptr)
+        {
+            buffer << "'" << wellKnownEventName << "' : ";
+        }
+        else
+        {
+            buffer << "'' : ";
+        }
+    }
+    buffer << id << "(" << eventId << ") v"
+           << version << "(" << eventVersion << ")";
+
+    if (numStackFrames > 0)
+    {
+        buffer << " [" << numStackFrames << " frames]";
+    }
+    buffer << std::endl;
+    std::cout << buffer.str();
+}
+
+void ClrEventsParser::ParseEvent(
+    EVENTPIPE_PROVIDER provider,
+    DWORD eventId,
+    DWORD eventVersion,
+    ULONG cbMetadataBlob,
+    LPCBYTE metadataBlob,
+    ULONG cbEventData,
+    LPCBYTE eventData,
+    LPCGUID pActivityId,
+    LPCGUID pRelatedActivityId,
+    ThreadID eventThread,
+    ULONG numStackFrames,
+    UINT_PTR stackFrames[])
+{
+    // Currently, only "Microsoft-Windows-DotNETRuntime" provider is used so no need to check.
+    // However, during the test, a last (k=0 id=1 V1) event is sent from "Microsoft-DotNETCore-EventPipe".
+    auto providerName = GetProviderName(provider);
+
+    // These should be the same as eventId and eventVersion.
+    // However it was not the case for the last event received from "Microsoft-DotNETCore-EventPipe".
+    DWORD id;
+    DWORD version;
+    INT64 keywords;  // used to filter out unneeded events.
+    WCHAR* name;
+    if (!TryGetEventInfo(metadataBlob, cbMetadataBlob, name, id, keywords, version))
+    {
+        return;
+    }
+
+    if (KEYWORD_GC == (keywords & KEYWORD_GC))
+    {
+        ParseGcEvent(id, version, cbEventData, eventData);
+    }
+    else
+    if (KEYWORD_CONTENTION == (keywords & KEYWORD_CONTENTION))
+    {
+        ParseContentionEvent(id, version, cbEventData, eventData);
+    }
+    else
+    {
+        // Dump to see all received events
+        //DumpEvent(providerName, name, id, version, keywords, eventId, eventVersion, numStackFrames);
+    }
+}
+
+
+// dump the buffer every 16 bytes + corresponding ASCII characters
+// ex:  44 4F 54 4E 45 54 5F 49  50 43 5F 56 31 00 77 00    DOTNET_IPC_V1.w.
+const DWORD LineWidth = 16;
```

Comment:
avoid defining global variables in the middle of the file. prefer static member (either class, either function)
scoping them is better.

Reply:
I'm removing the traces that were used in the console for investigation